data/security-posture.go

```diff
@@ -0,0 +1,60 @@
+package data
+
+import (
+	"github.com/google/go-github/v74/github"
+	"github.com/ossf/si-tooling/v2/si"
+)
+
+// SecurityPosture defines an interface for accessing security-related metadata about a repository.
+type SecurityPosture interface {
+	PreventsPushingSecrets() bool
+	ScansForSecrets() bool
+	DefinesPolicyForHandlingSecrets() bool
+}
+
+type RepoSecurityPosture struct {
+	restData                        RestData
+	preventsSecretPushing           bool
+	scansForSecrets                 bool
+	definesPolicyForHandlingSecrets bool
+}
+
+func buildSecurityPosture(repository *github.Repository, rd RestData) (SecurityPosture, error) {
+	securityConfig := repository.GetSecurityAndAnalysis()
+	if securityConfig == nil {
+		return &RepoSecurityPosture{
+			restData: rd,
+		}, nil
+	}
+	return &RepoSecurityPosture{
+		restData:              rd,
+		preventsSecretPushing: securityConfig.GetSecretScanning().GetStatus() == "enabled" || insightsClaimsSecretsTooling(rd.Insights),
+		scansForSecrets:       securityConfig.GetSecretScanning().GetStatus() == "enabled" || insightsClaimsSecretsTooling(rd.Insights),
+		// TODO: consider if SecurityInsights should have a policy doc field in ProjectDocumentation to handle this
+		// definesPolicyForHandlingSecrets: rd.SecurityInsights != nil && ....
+	}, nil
+}
```

Comment:
```suggestion
  secretsScanningStatus := securityConfig.GetSecretScanning().GetStatus()
  insightsClaimsSecretsTooling := insightsClaimsSecretsTooling(rd.Insights)
	return &RepoSecurityPosture{
		restData:              rd,
		preventsSecretPushing: secretsScanningStatus == "enabled" || insightsClaimsSecretsTooling,
		scansForSecrets:       secretsScanningStatus == "enabled" || insightsClaimsSecretsTooling,
		// TODO: consider if SecurityInsights should have a policy doc field in ProjectDocumentation to handle this
		// definesPolicyForHandlingSecrets: rd.SecurityInsights != nil && ....
	}, nil
}
```

Save results and compare instead of making multiple calls to same functions.